tests/dashboard/dashboard-teams.spec.js

```diff
@@ -27,17 +27,33 @@ const { LayersPanelPage } = require('../../pages/workspace/layers-panel-page');
 
 const maxDiffPixelRatio = 0.001;
 
+let teamPage,
+  loginPage,
+  registerPage,
+  dashboardPage,
+  layersPanelPage,
+  profilePage,
+  mainPage;
+
+mainTest.beforeEach(async ({ page }) => {
```

Comment:
It's really nice this refactor, much clear.
I doubt if it doesn't imply added time to the execution by instantiating pages not all the tests uses. 

I bet it's not considerable and it's much better this way, but we can take an eye on the execution times (once this PR is merged)  just to be aware of this fact.